src/app/forum/containers/thread/thread.component.ts

```diff
@@ -157,32 +169,76 @@ export class ThreadComponent implements AfterViewInit, OnDestroy {
     );
 
     this.subscriptions.add(
-      this.isThreadStatusOpened$.pipe(delay(0)).subscribe(isThreadStatusOpened => {
-        if (!isThreadStatusOpened) {
-          this.form.get('messageToSend')?.disable();
+      combineLatest([
+        this.threadStatus$.pipe(delay(0)),
+        this.disableControls$,
+      ]).subscribe(([ status, disableControls ]) => {
+        if (status?.data && (status.data.open || status.data.canOpen) && !disableControls) {
+          this.form.get('messageToSend')?.enable();
           return;
         }
-        this.form.get('messageToSend')?.enable();
-        this.messageToSendEl?.nativeElement.focus();
+        this.form.get('messageToSend')?.disable();
       })
     );
+    this.subscriptions.add(
+      this.isThreadStatusOpened$.pipe(delay(0), filter(isThreadStatusOpened => isThreadStatusOpened)).subscribe(() =>
+        this.messageToSendEl?.nativeElement.focus()
+      )
+    );
   }
 
   ngOnDestroy(): void {
     this.subscriptions.unsubscribe();
   }
 
-  sendMessage(threadId: ThreadId): void {
+  clearMessageToSendControl(): void {
+    this.form.reset({
+      messageToSend: '',
+    });
+  }
+
+  sendMessage(threadId: ThreadId, isThreadOpened: boolean): void {
     const messageToSend = this.form.value.messageToSend?.trim();
     if (!messageToSend) return;
-    this.store.select(forum.selectThreadToken).pipe(
+    this.disableControls$.next(true);
+
+    const threadToken$ = this.store.select(forum.selectThreadToken).pipe(
       take(1), // send on the current thread (if any) only
       filter(isNotUndefined),
-    ).subscribe(token => this.forumWebsocketClient.send(publishEventsAction(token, [ messageEvent(messageToSend) ])));
-    this.updateThreadService.update(threadId.itemId, threadId.participantId, { messageCountIncrement: 1 }).subscribe();
-    this.form.reset({
-      messageToSend: '',
+    );
+
+    if (isThreadOpened) {
+      threadToken$.subscribe({
+        next: token => this.forumWebsocketClient.send(publishEventsAction(token, [ messageEvent(messageToSend) ])),
+        error: () => this.disableControls$.next(false),
+      });
+      this.updateThreadService.update(threadId.itemId, threadId.participantId, { messageCountIncrement: 1 }).subscribe();
+      this.clearMessageToSendControl();
+      this.disableControls$.next(false);
+      return;
+    }
+
+    this.changeThreadStatus({
+      open: true,
+      threadId,
+      messageCountIncrement: 1,
+    }).subscribe({
+      error: () => this.disableControls$.next(false),
     });
+    this.subscriptions.add(
+      this.store.select(forum.selectThreadStatusOpen).pipe(
+        filter(isOpened => isOpened),
```

Comment:
is opened is not related with the thread open/close but just the websocket (common to all thread). Are you sure this component should know about WS state? looks strange (I haven't checked)

Reply:
Could you check please:

https://github.com/France-ioi/AlgoreaFrontend/blob/master/src/app/forum/store/current-thread/current-thread.selectors.ts#L45C9-L45C31
https://github.com/France-ioi/AlgoreaFrontend/blob/master/src/app/forum/models/threads.ts#L34

Reply:
of course, `selectThreadStatusOpen` is the thread one... I don't understand why I did this remark.